app/models/patron.rb

```diff
@@ -94,6 +94,16 @@ def member_list
     end
   end
 
+  def member_list_names
```

Comment:
I think something like this might be a more idiomatic way to write this (plus, with memoization)

```
@member_list_names ||= begin
  member_list.each_with_object({}) { |hash, member| hash[member.key] = member.first_name }
end
```

Reply:
Thanks for the suggestion!